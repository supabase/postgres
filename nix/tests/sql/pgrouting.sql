create schema v;

-- create the roads table
create table v.roads (
  id serial primary key,
  source integer,
  target integer,
  cost double precision
);

-- insert sample data into roads table
insert into v.roads (source, target, cost) values
(1, 2, 1.0),
(2, 3, 1.0),
(3, 4, 1.0),
(1, 3, 2.5),
(3, 5, 2.0);

-- create a function to use pgRouting to find the shortest path
select * from pgr_dijkstra(
  'select id, source, target, cost from v.roads',
  1, -- start node
  4  -- end node
);

drop schema v cascade;

