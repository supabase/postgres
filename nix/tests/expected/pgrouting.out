create schema v;
-- create the roads table
create table v.roads (
  id serial primary key,
  source integer,
  target integer,
  cost double precision
);
-- insert sample data into roads table
insert into v.roads (source, target, cost) values
(1, 2, 1.0),
(2, 3, 1.0),
(3, 4, 1.0),
(1, 3, 2.5),
(3, 5, 2.0);
-- create a function to use pgRouting to find the shortest path
select * from pgr_dijkstra(
  'select id, source, target, cost from v.roads',
  1, -- start node
  4  -- end node
);
 seq | path_seq | node | edge | cost | agg_cost 
-----+----------+------+------+------+----------
   1 |        1 |    1 |    1 |    1 |        0
   2 |        2 |    2 |    2 |    1 |        1
   3 |        3 |    3 |    3 |    1 |        2
   4 |        4 |    4 |   -1 |    0 |        3
(4 rows)

drop schema v cascade;
NOTICE:  drop cascades to table v.roads
